src/setup/SetupPanel.ts

```diff
@@ -325,6 +331,35 @@ export class SetupPanel {
         case "exploreComponents":
           await commands.executeCommand("esp.component-manager.ui.show");
           break;
+        case "canAccessFile":
+          if (message.path) {
+            const pathIdfPy = path.join(message.path, "tools", "idf.py");
+            const fileExists = await canAccessFile(pathIdfPy);
+            if (!fileExists) {
+              this.panel.webview.postMessage({
+                command: "canAccessFileResponse",
+                path: message.path,
+                exists: fileExists,
+              });
+            } else {
+              let versionEspIdf = message.currentVersion;
```

Comment:
Maybe we should validate that `message.currentVersion` does exist ?

Reply:
I agree! I've added the validation.